tests/components/backup/test_manager.py

```diff
@@ -2224,42 +2225,52 @@ async def test_receive_backup_file_read_error(
     assert unlink_mock.call_count == unlink_call_count
 
 
+@pytest.mark.usefixtures("path_glob")
 @pytest.mark.parametrize(
-    ("agent_id", "password", "restore_database", "restore_homeassistant", "dir"),
+    ("agent_id", "password_param", "restore_database", "restore_homeassistant", "dir"),
     [
-        (LOCAL_AGENT_ID, None, True, False, "backups"),
-        (LOCAL_AGENT_ID, "abc123", False, True, "backups"),
-        ("test.remote", None, True, True, "tmp_backups"),
+        (LOCAL_AGENT_ID, {}, True, False, "backups"),
+        (LOCAL_AGENT_ID, {"password": "abc123"}, False, True, "backups"),
+        ("test.remote", {}, True, True, "tmp_backups"),
     ],
 )
-async def test_async_trigger_restore(
+async def test_restore_backup(
     hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
     agent_id: str,
-    password: str | None,
+    password_param: dict[str, str],
     restore_database: bool,
     restore_homeassistant: bool,
     dir: str,
 ) -> None:
-    """Test trigger restore."""
-    manager = BackupManager(hass, CoreBackupReaderWriter(hass))
-    hass.data[DATA_MANAGER] = manager
-
-    await setup_backup_platform(hass, domain=DOMAIN, platform=local_backup_platform)
-    await setup_backup_platform(
-        hass,
-        domain="test",
-        platform=Mock(
-            async_get_backup_agents=AsyncMock(
-                return_value=[BackupAgentTest("remote", backups=[TEST_BACKUP_ABC123])]
+    """Test restore backup."""
+    password = password_param.get("password")
+    local_agent = local_backup_platform.CoreLocalBackupAgent(hass)
+    remote_agent = BackupAgentTest("remote", backups=[TEST_BACKUP_ABC123])
+    with patch(
+        "homeassistant.components.backup.backup.async_get_backup_agents"
+    ) as core_get_backup_agents:
```

Comment:
Why do patch here? Why can't we rely on the backup integration setting up the local backup agent?